network/connection.go

```diff
@@ -505,6 +505,9 @@ func (conn *Connection) SetVerdict(newVerdict Verdict, reason, reasonOptionKey s
 		if reasonOptionKey != "" && conn.Process() != nil {
 			conn.Reason.OptionKey = reasonOptionKey
 			conn.Reason.Profile = conn.Process().Profile().GetProfileSource(conn.Reason.OptionKey)
+		} else {
+			conn.Reason.OptionKey = ""
+			conn.Reason.Profile = ""
 		}
```

Comment:
```suggestion
              conn.Reason.OptionKey = reasonOptionKey
              conn.Reason.Profile = ""
		if process := conn.Process(); process != nil {
			conn.Reason.Profile = process.Profile().GetProfileSource(conn.Reason.OptionKey)
		}
```

Reply:
Though, I'm not sure `conn.Process()` can even return `nil`. Did you meant to check `conn.Process().Profile()` for `nil`?